[#1891] improvement(dev): dockerfile remove cache after installing packages

zhuzilong2013 opened the pull request:
### What changes were proposed in this pull request?

Remove cache after installing packages.

### Why are the changes needed?

Fix: #1891

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

By hand

justinmclean reviewed (dismissed):
LGTM

yuqi1129 commented:
@xunliu 
Can you assist in verifying this PR?

xunliu commented on dev/docker/hive/Dockerfile:
```diff
@@ -151,6 +151,10 @@ RUN chown -R datastrato:hadoop /home/datastrato
 
+################################################################################
+# removed install cache
+RUN rm -rf /var/cache/apt/*
```
hi @zhuzilong2013 Thank you for your contribution.

I think it's best to use the `apt clean` command to remove the cached files, but this needs to be tested.
https://askubuntu.com/questions/1350375/can-i-delete-var-cache-apt-archives

zhuzilong2013 replied:
Thanks for your comment. It's my mistake. I should use `rm -rf /var/lib/apt/list/*` actually.
In https://docs.docker.com/develop/develop-images/instructions/#run:
> Official Debian and Ubuntu images automatically run apt-get clean, so explicit invocation is not required.

I will fix and test it. I also want to move 'mysql-sever' to basic tools, what do you think?

yuqi1129 commented:
@zhuzilong2013 
Is it ready to be reviewed again?

zhuzilong2013 commented:
@yuqi1129 Please review it：）

yuqi1129 commented:
> @yuqi1129 Please review it：）

I'm okay with the changes, @jerryshao do you have any further comments?

qqqttt123 commented:
@xunliu @yuqi1129 Gently ping.

xunliu reviewed (commented):
hi @zhuzilong2013 Thank you for your update.
Have you tested to correctly create the Docker image using your development branch?

zhuzilong2013 commented:
hello @xunliu , I found the mysql silent installation failed because I moved 'mysql-sever' to basic tools. I added a new ARG to resolve it.
I build image on my M1 MacBook: 
<img width="1039" alt="截屏2024-02-20 18 02 52" src="https://github.com/datastrato/gravitino/assets/22188816/1fbe01fb-76bf-44fc-bb0f-a2d8d8647ed3">
Please review it.

xunliu commented on dev/docker/hive/Dockerfile:
```diff
@@ -9,6 +9,7 @@ LABEL maintainer="support@datastrato.com"
 ARG HADOOP_PACKAGE_NAME
 ARG HIVE_PACKAGE_NAME
 ARG JDBC_DIVER_PACKAGE_NAME
+ARG DEBIAN_FRONTEND=noninteractive
```
Where use this `DEBIAN_FRONTEND` arg? I didn't find it anywhere.

zhuzilong2013 replied:
I added it because I moved 'mysql-sever' to basic tools.

xunliu replied:
OK, I found `DEBIAN_FRONTEND` it in the
https://askubuntu.com/questions/972516/debian-frontend-environment-variable

xunliu reviewed (approved):
LGTM

mchades commented:
Has this PR been tested with IT? @xunliu @zhuzilong2013
